Added tests for Deprovision and Unbind.

SamiSousa opened the pull request:
Also added test for whitelist entries.
Added Ping to the Provision step.

danmcp commented on tests/broker_test.go:
```diff
@@ -53,12 +59,65 @@ func TestBrokerLogic(t *testing.T){
+	// Instance doesn't exist
+	_, errProvisionFake := businessLogic.Provision(
+		&osb.ProvisionRequest{
+			InstanceID:	"not-an-instance",
```
Formatting is a mess here.  Has this run through go format?  Ideally a format check should be part of travis.

danmcp commented on tests/whitelist_test.go:
```diff
@@ -0,0 +1,83 @@
+			uuids[dataverse.PlanID] = 1
+		}
+
+		// Existance
```
Spelling

danmcp commented on tests/whitelist_test.go:
```diff
@@ -0,0 +1,83 @@
+	- Completeness: Has all required info for a dataverse service
+		(server name, identifier, Name, etc)
+	- Uniqueness: Unique service ids and plan ids, etc
+	- Existance: Pings the server to see if the dataverse exists/is live
```
Spelling

danmcp commented:
@SamiSousa Please squash these commits.

SamiSousa commented:
Fixed typos. Added pre-commit hook for automatic go formatting on commits. Formatted all code in pkg/broker and tests.
We'll squash when we merge this into master.

danmcp commented:
@SamiSousa Little confused about your squash statement.  Why wouldn't you squash it in the PR?  I assume you will be merging the PR to get it into master.

danmcp commented:
@SamiSousa Also to be clear, it doesn't have to squashed into 1 commit because you have multiple authors.  But at least remove in intermediate state commits.

SamiSousa commented:
Actually the changes are already merged to master, I erroneously left them on my branch.

SamiSousa commented:
Added a format check to Travis, and ran gofmt on all go files. IShould I also include a script to automatically format all go code within the repo?

ckl2150 commented on pkg/broker/types.go:
```diff
@@ -19,72 +19,69 @@ type BusinessLogic struct {
 // type for JSON portion describing a dataverse on Server
-type DataverseDescription struct{
+type DataverseDescription struct {
 	// Wow, capitalization matters for structs in go?
```
Can probably take this out now heh

SamiSousa replied:
Nice catch. I'll clean up my comments

ckl2150 commented:
Other than inclusion of gofmt, looks like a lot of quality-of-life changes (formatting, etc). Looks good to me.

SamiSousa commented:
Cleaned up comments
